### ISSUE TITLE
Nullability issues when serializable class is not explicitly annotated

### ISSUE THREAD

**Issue by suushiemaniac** [pdvrieze/xmlutil#31]
Kotlin: 1.3.72 JVM
KotlinX serialization runtime: 0.20.0
xmlutil: 0.20.0.1

When de-serializing XML into a class that has nullable attributes, the parsing fails under certain circumstances. Example:

```kotlin
@Serializable
// uncomment this annotation as quick-fix @XmlSerialName("XmlRec", "", "")
data class XmlRec(val item: Int, val maybeRec: XmlRec? = null)

val thisExplodes = XmlRec(3, XmlRec(2, XmlRec(1))) // inner-most entity implicitly has maybeRec = null

val namespaceLazySerial = XML { repairNamespaces = false }

val explosiveXml = namespaceLazySerial.stringify(XmlRec.serializer(), thisExplodes)
val explosion = namespaceLazySerial.parse(XmlRec.serializer(), explosiveXml)
```

With the `XmlSerialName` annotation commented out, de-serializing fails with the following error message: `nl.adaptivity.xmlutil.serialization.UnknownXmlFieldException: Could not find a field for name XmlRec candidates: item, XmlRec? at position Line number = 2`

Serializing works as expected. By the looks of the error message it seems that the internal code does a string comparison between `XmlRec` and `XmlRec?` somewhere, which of course turns out false. I would reckon however that with de-serializing it seems reasonable to use the `XmlRec?` deserialization strategy because the XML parser has already encountered an `<XmlRec>` tag (that he is trying to unserialize unsuccessfully, therefor prompting the error message) that will arguably not be null anyways.

Interestingly, with the `XmlSerialName` annotation in place everything works because we then force the parser to not infer the nullability `?` in the deserialization strategy name.

**Comment by pklimai** [comment]
I got into a similar issue, and can confirm that `@XmlSerialName` is really a workaround. We are currently using it but would be good to know if it is a bug or feature.

**Comment by sgarfinkel** [comment]
Just got hit with this too. Thanks for the info for the workaround, but this is definitely a bug.

**Comment by pdvrieze** [comment]
I guess this is due to the change of the serialization library. I'll fix it, but the workaround should work for now.

**Comment by pdvrieze** [comment]
Just fixed in c24b585

**Comment by pdvrieze** [comment]
Now released in v0.20.0.10
